From 505130447a4a0b356ada0cc8f223b4330e4c2e40 Mon Sep 17 00:00:00 2001
From: Zhiyi Zhang <zzhang@codeweavers.com>
Date: Mon, 23 Jun 2025 17:24:28 +0800
Subject: [PATCH] Revert "wineps: Use the correct colours when a monochrome
 bitmap without a colour table is the source."

This reverts commit ad7e889c1d58aec091e72f08db86346c74070128.

Manual tests show that the color table is always used.
---
 dlls/wineps.drv/bitmap.c | 18 ++++--------------
 1 file changed, 4 insertions(+), 14 deletions(-)

diff --git a/dlls/wineps.drv/bitmap.c b/dlls/wineps.drv/bitmap.c
index 74a18c5344b..ccb66eb64c0 100644
--- a/dlls/wineps.drv/bitmap.c
+++ b/dlls/wineps.drv/bitmap.c
@@ -108,20 +108,10 @@ static BOOL PSDRV_WriteImageMaskHeader(print_ctx *ctx, const BITMAPINFO *info, I
        the foregnd color corresponds to a bit equal to
        0 in the bitmap.
     */
-    if (!info->bmiHeader.biClrUsed)
-    {
-        PSDRV_CreateColor( ctx, &foregnd, GetTextColor( ctx->hdc ) );
-        bkgnd = ctx->bkColor;
-    }
-    else
-    {
-        PSDRV_CreateColor( ctx, &foregnd, RGB(info->bmiColors[0].rgbRed,
-                                              info->bmiColors[0].rgbGreen,
-                                              info->bmiColors[0].rgbBlue) );
-        PSDRV_CreateColor( ctx, &bkgnd, RGB(info->bmiColors[1].rgbRed,
-                                            info->bmiColors[1].rgbGreen,
-                                            info->bmiColors[1].rgbBlue) );
-    }
+    PSDRV_CreateColor(ctx, &foregnd, RGB(info->bmiColors[0].rgbRed, info->bmiColors[0].rgbGreen,
+                      info->bmiColors[0].rgbBlue));
+    PSDRV_CreateColor(ctx, &bkgnd, RGB(info->bmiColors[1].rgbRed, info->bmiColors[1].rgbGreen,
+                      info->bmiColors[1].rgbBlue));
 
     PSDRV_WriteGSave(ctx);
     PSDRV_WriteNewPath(ctx);
-- 
GitLab

